### PR TITLE
Fix rpcv02 tests

### PR DESCRIPTION
**Pull request by rianhughes**
https://github.com/NethermindEth/starknet.go/issues/217

### CODE REVIEW

**Comment by cicr99** [comment]
Some tests are still failing for me on devnet:
- `TestCall` on `call_test.go`
- `TestChainID` on `provider_test.go`
- `TestDeclareTransaction` on `write_test.go`

**Comment by cicr99** [comment]
Also could we update the workflows? `Testing RPC v0.2 on devnet` is being skipped atm. In addition, testing on `testnet` is running but is failing, we could skip this step until we manage to fix them as well.
Would love to see a green status

**Comment by rianhughes** [comment]
All tests should pass on mainnet, testnet, devnet, and mock, when making requests to Juno

File: rpcv02/chain.go [hunk 1]
```diff
@@ -22,11 +23,18 @@ func (provider *Provider) ChainID(ctx context.Context) (string, error) {
...
-	var result SyncStatus
+	var result interface{}
 	// Note: []interface{}{}...force an empty `params[]` in the jsonrpc request
 	if err := provider.c.CallContext(ctx, &result, "starknet_syncing", []interface{}{}...); err != nil {
 		return nil, err
 	}
-	return &result, nil
+	switch res := result.(type) {
+	case bool:
+		return &SyncStatus{SyncStatus: res}, nil
```
Comment (cicr99) [inline rpcv02/chain.go:33]: Not sure if this falls under rpc v0.2 scope but the only possible value for `result` when it has bool type is `False`. Should we check that here?

File: rpcv02/types.go [hunk 2]
```diff
@@ -19,15 +19,15 @@ type StorageEntry struct {
 	Value *felt.Felt `json:"value"`
 }
 
-type StorageEntries struct {
-	StorageEntry []StorageEntry
-}
+// type StorageEntries struct {
+// 	StorageEntry []StorageEntry
+// }
```
Comment (cicr99) [inline rpcv02/types.go:24]: This can be removed now, right?

File: rpcv02/call_test.go [hunk 3]
```diff
@@ -2,90 +2,82 @@ package rpcv02
...
 		},
 		"testnet": {
 			{
-				FunctionCall: types.FunctionCall{
-					ContractAddress:    types.StrToFelt("0x029260ce936efafa6d0042bc59757a653e3f992b97960c1c4f8ccd63b7a90136"),
-					EntryPointSelector: "decimals",
-					Calldata:           []string{},
+				FunctionCall: FunctionCall{
+					ContractAddress:    utils.TestHexToFelt(t, "0x029260ce936efafa6d0042bc59757a653e3f992b97960c1c4f8ccd63b7a90136"),
+					EntryPointSelector: utils.TestHexToFelt(t, "0x004c4fb1ab068f6039d5780c68dd0fa2f8742cceb3426d19667778ca7f3518a9"), // types.GetSelectorFromNameFelt("decimals"), // 0x004c4fb1ab068f6039d5780c68dd0fa2f8742cceb3426d19667778ca7f3518a9
```
Comment (cicr99) [inline rpcv02/call_test.go]: what about these comments?

File: rpcv02/chain.go (hunk 1 above)
Comment (rianhughes) [reply rpcv02/chain.go:33]: I don't think we need to as the endpoint will (/should) never return True

File: rpcv02/types.go (hunk 2 above)
Comment (rianhughes) [reply rpcv02/types.go:24]: Yes

**Review by cicr99 — APPROVED** [review]
lgtm

**Pull request merged**